wandb/sdk/artifacts/storage_handlers/fsspec_file_handler.py

```diff
@@ -0,0 +1,146 @@
+
+        Returns:
+            (os.PathLike): A path to the file represented by `index_entry`
+        """
+        if not local:
+            assert manifest_entry.ref is not None
+            return manifest_entry.ref
+
+        if manifest_entry.ref is None:
+            raise ValueError(f"Cannot add path with no ref: {manifest_entry.path}")
+        fs, fs_path = fsspec.core.url_to_fs(str(manifest_entry.ref))
+
+        if not fs.exists(fs_path):
+            raise ValueError(
+                "fsspec file reference: Failed to find file at path %s" % fs_path
+            )
+
+        path, hit, cache_open = self._cache.check_md5_obj_path(
+            B64MD5(manifest_entry.digest),  # TODO(spencerpearson): unsafe cast
+            manifest_entry.size if manifest_entry.size is not None else 0,
+        )
+        if hit:
+            return path
+
+        md5 = fs.checksum(fs_path)
+        if md5 != manifest_entry.digest:
+            raise ValueError(
+                f"fsspec file reference: Digest mismatch for path {manifest_entry.path}: expected {manifest_entry.digest} but found {md5}"
+            )
+
+        filesystem.mkdir_exists_ok(os.path.dirname(path))
+        fs.get_file(fs_path, path)
+
+        return path
+
+    def store_path(
+        self,
+        artifact: "Artifact",
+        path: Union[URIStr, FilePathStr],
+        name: Optional[StrPath] = None,
+        checksum: bool = True,
+        max_objects: Optional[int] = None,
+    ) -> Sequence[ArtifactManifestEntry]:
+        max_objects = max_objects or DEFAULT_MAX_OBJECTS
+        # We have a single file or directory
+        entries = []
+        fs, fs_path = fsspec.core.url_to_fs(path)
+
+        def md5(path: str) -> Any:
+            return (
+                fs.checksum(path)
+                if checksum
+                else md5_string(str(fs.info(path)["size"]))
+            )
+
+        if fs.isdir(fs_path):
+            i = 0
+            start_time = time.time()
+            if checksum:
+                termlog(
+                    'Generating checksum for up to %i files in "%s"...\n'
+                    % (max_objects, fs_path),
+                    newline=False,
+                )
+            for root, _, files in fs.walk(fs_path):
+                for sub_path in files:
+                    i += 1
+                    if i > max_objects:
+                        raise ValueError(
+                            "Exceeded %i objects tracked, pass max_objects to add_reference"
+                            % max_objects
+                        )
+                    physical_path = os.path.join(root, sub_path)
+                    relative_path = os.path.relpath(physical_path, start=fs_path)
+                    if name is not None:
+                        relative_path = os.path.join(name, relative_path)
+
+                    entry = ArtifactManifestEntry(
+                        path=relative_path,
```

Comment:
Wrap in `LogicalPath` to replace `\` with `/` on Windows:
```suggestion
                        path=LogicalPath(relative_path),
```

Reply:
Good remark, also adjusted this in `local_file_handler.py`